### PR TITLE
fix cpp_info.names and cpp_info.filenames in the local flow

### LINKED ISSUE #7854: [bug] deps_cpp_info.get_name returns different results with `create` and `build`
### Environment Details (include every applicable attribute)
  * Operating System+version: CentOS 7
  * Compiler+version: gcc 4.8.5
  * Conan version: 1.30.0
  * Python version: 3.6.8

### Steps to reproduce (Include if Applicable)

The following conanfile.py

```
import conans

class Package(conans.ConanFile):
	name = 'package-name'
	version = '0.0.1'
	requires = 'gtest/1.10.0'
	generators = 'cmake_find_package'

	def build(self):
		print(self.deps_cpp_info['gtest'].get_name('cmake_find_package'))
```

prints different gtest package name depending on which conan commands I use:

`conan create .` prints `GTest`
`conan install . && conan build .` prints `gtest`

I expect the second command to print `GTest` like the first one.

### PR DESCRIPTION
**Pull request by memsharded**
Changelog: Bugfix: Fix local flow (conan install + build) support for ``cpp_info.names`` and ``cpp_info.filenames``.
Docs: Omit

Close https://github.com/conan-io/conan/issues/7854

### CODE REVIEW

**Review by danimtb — APPROVED** [review]

File: conans/client/generators/text.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def __init__(self, cpp_info):
         self.version = cpp_info.version
         self.name = cpp_info.get_name(TXTGenerator.name)
         self.rootpath = "%s" % cpp_info.rootpath.replace("\\", "/")
+        self.generatornames = "\n".join("%s:%s" % (k, v) for k, v in cpp_info.names.items())
+        self.generatorfilenames = "\n".join("%s:%s" % (k, v) for k, v in cpp_info.filenames.items())
```
Comment (jgsogo) [inline conans/client/generators/text.py]: Better to use `=` as separator (more ini-like format)? Or is there any reason to use `:`?

File: conans/client/generators/text.py [hunk 2]
```diff
@@ -164,7 +166,17 @@ def _populate_cpp_info(_cpp_info, _data, _rootpath):
                 rootpath = no_config_data.pop('rootpath')[0]
                 dep_cpp_info = CppInfo(dep, rootpath)
                 dep_cpp_info.filter_empty = filter_empty
-                dep_cpp_info.names[TXTGenerator.name] = no_config_data.pop('name')[0]
+                _ = no_config_data.pop('name')[0]
+                version = no_config_data.pop('version', [""])[0]
+                dep_cpp_info.version = version
+                generatornames = no_config_data.pop("generatornames", [])  # can be empty
+                for n in generatornames:
+                    gen, value = n.split(":")
```
Comment (jgsogo) [inline conans/client/generators/text.py]:
```suggestion
                    gen, value = n.split(":", 1)
```

File: conans/client/generators/text.py [hunk 3]
```diff
@@ -164,7 +166,17 @@ def _populate_cpp_info(_cpp_info, _data, _rootpath):
...
+                    dep_cpp_info.names[gen] = value
+                generatorfilenames = no_config_data.pop("generatorfilenames", [])  # can be empty
+                for n in generatorfilenames:
+                    gen, value = n.split(":")
```
Comment (jgsogo) [inline conans/client/generators/text.py]:
```suggestion
                    gen, value = n.split(":", 1)
```

File: conans/test/unittests/client/generators/txt/test_dump_load.py [hunk 4]
```diff
@@ -17,17 +17,16 @@ def test_names_per_generator(self):
         cpp_info = CppInfo("pkg_name", "root")
         cpp_info.name = "name"
         cpp_info.names["txt"] = "txt_name"
-        cpp_info.names["cmake_find_package"] = "cmake_find_package"
+        cpp_info.names["cmake_find_package"] = "SpecialName"
```
Comment (jgsogo) [inline conans/test/unittests/client/generators/txt/test_dump_load.py:20]: Add `cpp_info.filenames` here too, we can check the other field as well.

**Review by jgsogo — APPROVED** [review]

**Pull request merged**